snuba/datasets/metrics_messages.py

```diff
@@ -16,6 +16,7 @@ class OutputType(Enum):
 
 class AggregationOption(Enum):
     HIST = "hist"
+    TEN_SECOND = "ten_second"
```

Comment:
This seems it would make the configuration incongruent. Why is 10 second special enough to have its own flag while the others do not? 
Possible options:
- Have a flag per granularity so as I can turn on and off any granularity, not just the 10 seconds.
- Have a structured config where we have a field "granularity" that is a set of granularities that I want.

Reply:
I think the reason why 10 second granularity is singled out in this way is because the other granularities are all enabled by default, but 10s is not. 

But yea, I think this makes sense. I can try to reconfigure this (might go with option #2)

Reply:
@fpacifici Originally this `AggregationOption` thing was designed to be highly opinionated. The enums are suppose to be things like `ALERT` etc. Where snuba would parse it and apply the correct aggregations (for the case of `ALERT` it would be only producing a `[0]` in the `granularities` column and then set the retention to 7 days or less). But for various reasons, those requirements never came.

@nikhars I'm thinking of rebuilding this thing since people using it in a clearly different way than we intended. I think perhaps we could tie in with your idea of "having a better metrics use case creation experience" somehow.

Reply:
I agree that the system for allowing different types of configurations needs a refactor. What we did earlier was very specific focusing on one detail only. If we want to allow more types of configurations to be possible, we should make the workflows more generic.

Reply:
(I asked a similar question below, but) Do we have a preference for the order in which this work should be done? Do we think that this could be follow-up work after we unlock 10s granularity for the relevant product team?

Reply:
I think you could keep the configuration congruent and consistent without redesigning the whole configuration schema. This could be one by adding one flag per granularity interval. At least this makes the 10 seconds granularity not special. 
Then you can rearchitect it with a better overall schema separately.

@john-z-yang, I like the idea of having configuration templates. But it must be possible to introduce new ones easily, otherwise you will be in a position where the platform would need a lot of product knowledge to be useful.

Reply:
Makes sense. I can add flags so that 10s granularity isn't a one-off in the configuration.

Reply:
Thanks, @fpacifici, how would you feel about
1. Sentry sending aggregation options in a list instead of a single one like I originally planned. Say if someone wants both histogram and 10 sec granularity, they could just add it to the list, so we do not get an explosion in this enum.
2. As for having a flag per granularity, I think for now we should only have a flag if it is disabled by default, and the existence of the flag means to enable it? It doesn't make a lot of sense to me why we want the other granularities as flag as that would imply there would be a product need to disable them.

cc @nikhars @ayirr7 on your thoughts

Reply:
> I think for now we should only have a flag if it is disabled by default, and the existence of the flag means to enable it? It doesn't make a lot of sense to me why we want the other granularities as flag as that would imply there would be a product need to disable them

With this approach how would you disable retentions that are enabled by default ? 
You have three options:
- treat everything as disabled by default and add enabled flags 
- the opposite and treat everything as enabled by default and have a flag to disable
- Have negative flags for enabled by default and positive flags for disabled by default.

I htink both 1 and 2 are ok. I like 3 less as it adds cognitive load in knowhing which flags are positive and which ones are negative.